src/mesh.c

```diff
@@ -0,0 +1,426 @@
+/**
+ * collectd - src/mesh.c
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; only version 2 of the License is applicable.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * Authors:
+ *   Pierre Lebleu <pierre.lebleu at pile-engineering.com>
+ */
+
+#include "collectd.h"
+#include <inttypes.h>
+
+#include "plugin.h"
+#include "utils/common/common.h"
+
+#if !KERNEL_LINUX
+#error "This module only supports the Linux implementation of ieee8021"
+#endif
+
+#define PLUGIN_NAME "mesh"
+
+#define SYS_PATH "/sys/kernel/debug/ieee80211/"
+
+/*
+ * Device attributes
+ */
+#define ESTAB_PLINKS "estab_plinks" // The number of established peer links
+#define DROPPED_FRAMES_CONGESTION                                              \
+  "dropped_frames_congestion" // The number of dropped frames due to congestion
+#define DROPPED_FRAMES_NO_ROUTE                                                \
+  "dropped_frames_no_route" // The number of dropped frames due to unrouteable
+#define DROPPED_FRAMES_TTL                                                     \
+  "dropped_frames_ttl" // The number of dropped frames due to TTL
+#define FORWARDED_FRAMES "fwded_frames" // The number of forwarded frames
+#define FORWARDED_UNICAST                                                      \
+  "fwded_unicast" // The number of unicast forwarded frames
+#define FORWARDED_MCAST                                                        \
+  "fwded_mcast" // The number of multicast forwarded frames
+
+/*
+ * Metric labels
+ */
+#define LABEL_NAME "name"
+#define LABEL_REASON "reason"
+#define LABEL_TYPE "type"
+
+/*
+ * The config key strings
+ */
+#define PHY_DEVICE_KEY "PhysicalDevice"
+#define NET_DEVICE_KEY "NetworkDevice"
+
+#define PHY_NAME_LEN 32
+#define NET_NAME_LEN 32
+
+struct mesh_list {
+  char phy_name[PHY_NAME_LEN];
+  char net_name[NET_NAME_LEN];
+  struct mesh_list *next;
+}; /* mesh_list */
+typedef struct mesh_list mesh_list_t;
+
+typedef enum {
+  MESH_FAM_DROPPED,
+  MESH_FAM_FORWARDED,
+  MESH_FAM_PEERS,
+  MESH_FAM_MAX,
+} mesh_family_t;
+
+static metric_family_t g_families[MESH_FAM_MAX] = {
+    [MESH_FAM_DROPPED] =
+        {
+            .name = "hw.network.packets.dropped",
+            .help = "",
```

Comment:
Either have a help description, or drop the `.help` members.   Write plugins should handle NULL members just fine (Prometheus plugin does, if others do not, that's a bug).